### PR TITLE
Fix Blueprint name mismatch

### LINKED ISSUE #2542: [blueprints] No error message when Blueprint name doesn't match instance name in yaml
Given [the blueprint](https://github.com/canonical/multipass-blueprints/blob/add-fake-bprint/v1/fake2.yaml):
```yaml
description: A Fake blueprint to test CI
version: latest

instances:
  fake:
    image: 22.04
    limits:
      min-mem: 1G
    cloud-init:
      vendor-data: |
        packages: [cowsay]
health-check: |
  cowsay fake2
```

Multipass launches that with `-m 0M`:
- on `qemu`:
  `$ mp launch fake2 -vvvv`
  `# ...`
<tt>[2022-04-29T05:27:31.820] [debug] [qemu-system-x86_64] [759060] started: qemu-system-x86_64 --enable-kvm -cpu host -nic tap,ifname=tap-e7e9518ba87,script=no,downscript=no,model=virtio-net-pci,mac=52:54:00:8a:73:a2 -device virtio-scsi-pci,id=scsi0 -drive file=/var/snap/multipass/common/data/multipassd/vault/instances/fake2/ubuntu-20.04-server-cloudimg-amd64.img,if=none,format=qcow2,discard=unmap,id=hda -device scsi-hd,drive=hda,bus=scsi0.0 -smp 1 <strong>-m 0M</strong> -qmp stdio -chardev null,id=char0 -serial chardev:char0 -nographic -cdrom /var/snap/multipass/common/data/multipassd/vault/instances/fake2/cloud-init-config.iso</tt>
  `[2022-04-29T05:27:31.820] [info] [fake2] process started`
- on `lxd`:
  `$ mp launch fake2 -vvvv`
  `# ...`
  `launch failed: Operation completed with error: (400) Failed creating instance record: Memory limit can't be 0`

```ini
# /etc/systemd/system/snap.multipass.multipassd.service.d/override.conf
[Service]
Environment="MULTIPASS_BLUEPRINTS_URL=https://github.com/canonical/multipass-blueprints/archive/refs/pull/17/merge.zip"
ExecStart=
ExecStart=/usr/bin/snap run multipass.multipassd --verbosity trace
```

### PR DESCRIPTION
**Pull request by luis4a0**
Fixes #2542.

### CODE REVIEW

File: src/blueprint_provider/default_vm_blueprint_provider.cpp [hunk 1]
```diff
@@ -110,6 +110,11 @@ mp::Query mp::DefaultVMBlueprintProvider::fetch_blueprint_for(const std::string&
     Query query{"", "default", false, "", Query::Type::Alias};
     auto& blueprint_config = blueprint_map.at(blueprint_name);
 
+    if (!blueprint_config["instances"][blueprint_name])
+    {
+        throw InvalidBlueprintException(fmt::format("Definition of \"{}\" not found", blueprint_name));
```
Comment (townsend2010) [inline src/blueprint_provider/default_vm_blueprint_provider.cpp:115]: I think having something like
```
fmt::format("There are no instance definitions matching Blueprint name \"{}\"", blueprint_name));
```
makes it a bit more clear what the error is.

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Hey @luis4a0,

Looks good, thanks!  One little suggestion inline and I think it's good.

File: src/blueprint_provider/default_vm_blueprint_provider.cpp (hunk 1 above)
Comment (luis4a0) [reply src/blueprint_provider/default_vm_blueprint_provider.cpp]: Right. Done.

**Review by townsend2010 — APPROVED** [review]
LGTM, thanks!

bors merge

**Comment by townsend2010** [comment]
There was an issue with the action-runners on the Mac.  Will try one more time.

bors retry

**Comment by townsend2010** [comment]
Failed an integration test.  Will merge manually...

**Pull request merged**